pages/topics/oauth2.mdx

```diff
@@ -26,84 +32,173 @@ In accordance with the relevant RFCs, the token and token revocation URLs will *
+Before we dive into the semantics of the different OAuth2 grants, we should stop and discuss security, specifically the use of the `state` parameter.
+[Cross Site Request Forgery](https://en.wikipedia.org/wiki/Cross-site_request_forgery), or CSRF, and [Clickjacking](https://en.wikipedia.org/wiki/Clickjacking) are security vulnerabilities that must be addressed by individuals implementing OAuth.
+This is typically accomplished using the `state` parameter. `state` is sent in the authorization request and returned back in the response and should be a value that binds the user's request to their authenticated state.
+For example, `state` could be a hash of the user's session cookie, or some other nonce that can be linked to the user's session.
+
+When a user begins an authorization flow on the client, a `state` is generated that is unique to that user's request. This value is stored somewhere only accessible to the client and the user, i.e. protected by the [same-origin policy](https://en.wikipedia.org/wiki/Same-origin_policy).
+When the user is redirected, the `state` parameter is returned. The client validates the request by checking that the `state` returned matches the stored value. If they match, it is a valid authorization request.
+If they do not match, it's possible that someone intercepted the request or otherwise falsely authorized themselves to another user's resources, and the request should be denied.
 
-When a user begins an authorization flow on the client, a `state` is generated that is unique to that user's request. This value is stored somewhere only accessible to the client and the user, i.e. protected by the [same-origin policy](https://en.wikipedia.org/wiki/Same-origin_policy). When the user is redirected, the `state` parameter is returned. The client validates the request by checking that the `state` returned matches the stored value. If they match, it is a valid authorization request. If they do not match, it's possible that someone intercepted the request or otherwise falsely authorized themselves to another user's resources, and the request should be denied.
+For OpenID Connect, the `nonce` parameter may be used in a similar way instead. Upon receiving the ID token, the client should validate that the `nonce` claim in the ID token matches the `nonce` sent in the authorization request.
 
-While Discord does not require the use of the `state` parameter, we support it and highly recommend that you implement it for the security of your own applications and data.
+While Discord does not require the use of the `state` parameter, it supports it and we highly recommend that you implement it for the security of your own applications and data.
```

Comment:
```suggestion
While Discord does not require the use of the `state` parameter, it is supported and we highly recommend that you implement it for the security of your own applications and data.
```

Reply:
better to just remove the phrase entirely